### PR TITLE
Create backend API for admin page.

### PR DESCRIPTION
**Pull request by rbclark**
This should not affect the current admin page, however this pull request will add a backend API with tests for managing users via an admin panel.

### CODE REVIEW

File: config/initializers/apipie.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
   config.api_base_url            = "/api"
   config.doc_base_url            = "/apipie"
   config.default_version         = "3.0"
-  config.api_controllers_matcher = "#{Rails.root}/app/controllers/api/*.rb"
+  config.api_controllers_matcher = ["#{Rails.root}/app/controllers/api/*.rb", "#{Rails.root}/app/controllers/api/admin/*.rb"]
```
Comment (schreiaj) [inline config/initializers/apipie.rb:6]: @eedrummer If this goes through the Ruby Dir.glob function can't you use `"#{Rails.root}/app/controllers/api/**.rb"` instead?

Comment (rbclark) [inline config/initializers/apipie.rb:6]: Oops, yeah that would work fine as well. Updated and pushed again.

**Comment by schreiaj** [comment]
Holding off on merging this until /api/patients/count is moved to /api/admin/patients and cache and provider have some way of retrieving counts on them.

File: config/initializers/apipie.rb (hunk 1 above)
Comment (rbclark) [inline config/initializers/apipie.rb:6]: Actually had to revert this back to the original way I had it, it was causing apipie to no longer show the admin panel documentation.

**Comment by rbclark** [comment]
Okay the additional required routes have been added and tests have been fixed.

**Pull request merged**